### PR TITLE
Add audience management

### LINKED ISSUE #271: Support Audience apis
News: 
https://developers.line.biz/en/news/2020/02/03/release-audience-sharing/ 
https://developers.line.biz/en/news/2020/03/04/all-narrowcast-english/
Doc: https://developers.line.biz/en/docs/messaging-api/using-audience/

- [ ] [Create audience for uploading user IDs](https://developers.line.biz/en/reference/messaging-api/#create-upload-audience-group)
- [ ] [Create audience for click-based retargeting](https://developers.line.biz/en/reference/messaging-api/#create-click-audience-group)
- [ ] [Create audience for impression-based retargeting](https://developers.line.biz/en/reference/messaging-api/#create-imp-audience-group)
- [ ] [Rename an audience](https://developers.line.biz/en/reference/messaging-api/#set-description-audience-group)
- [ ] [Delete audience](https://developers.line.biz/en/reference/messaging-api/#delete-audience-group)
- [ ] [Get audience data](https://developers.line.biz/en/reference/messaging-api/#get-audience-group)
- [ ] [Get data for multiple audiences](https://developers.line.biz/en/reference/messaging-api/#get-audience-groups)
- [ ] [Get the authority level of the audience](https://developers.line.biz/en/reference/messaging-api/#get-audience-groups)
- [ ] [Change the authority level of the audience](https://developers.line.biz/en/reference/messaging-api/#change-authority-level)

### PR DESCRIPTION
**Pull request by moririnson**
fixes #271

### CODE REVIEW

File: src/LINEBot/HTTPClient/CurlHTTPClient.php [hunk 1]
```diff
@@ -111,7 +125,7 @@ private function getOptions($method, $headers, $reqBody)
             CURLOPT_BINARYTRANSFER => true,
             CURLOPT_HEADER => true,
         ];
-        if ($method === 'POST') {
+        if ($method === 'POST' || $method === 'PUT') {
```
Comment (satosby) [inline src/LINEBot/HTTPClient/CurlHTTPClient.php:128]: It seems that CURLOPT_PUT is not set

File: tests/LINEBot/ManagementAudienceTest.php [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+            /** @var \PHPUnit\Framework\TestCase $testRunner */
+            $testRunner->assertEquals('GET', $httpMethod);
+            $testRunner->assertEquals('https://api.line.me/v2/bot/audienceGroup/authorityLevel', $url);
+
+            return [
+                'authorityLevel' => 'PUBLIC',
+            ];
+        };
+        $bot = new LINEBot(new DummyHttpClient($this, $mock), ['channelSecret' => 'CHANNEL-SECRET']);
+        $res = $bot->getAuthorityLevel();
```
Comment (satosby) [inline tests/LINEBot/ManagementAudienceTest.php:470]: Is it updateAuthorityLevel instead of getAuthorityLevel?

Comment (moririnson) [reply tests/LINEBot/ManagementAudienceTest.php]: yes. fixed~

File: src/LINEBot/HTTPClient/CurlHTTPClient.php (hunk 1 above)
Comment (moririnson) [reply src/LINEBot/HTTPClient/CurlHTTPClient.php]: unnecessary to use `CURLOPT_PUT ` because `CURLOPT_CUSTOMREQUEST => $method` used.

**Pull request merged**